### DISCUSSION TITLE
django-fsm-2

### DISCUSSION THREAD

**Discussion by tim-schilling** [django-commons/membership#79] [Check-ins]
Hi @django-commons/django-fsm-2, I wanted to check-in with you about your project, logistics and any concerns you may have.

1. Have you run into any friction having your project in Django Commons, if so, can you describe them?
2. Is there anything we can do to help your team, repo out?
3. Do you have any questions or concerns for us?

If you'd like to chat with us privately, our email is django-commons-admin@googlegroups.com

**Comment by Natim** [comment]
No frictions, no help needed and no questions so far, thanks :pray:

**Reply by tim-schilling** [reply]
Good stuff! We'll circle back again in ~6-9 months. Though if anything comes up, let us know.

**Comment by tim-schilling** [comment]
Hi @django-commons/django-fsm-2, I'm circling back around :dizzy: In addition to my original questions, we're looking into how can we improve the new contributor experience with the overall goal of helping find regular contributor and more maintainers for projects.
- [Meet and greets](https://github.com/orgs/django-commons/discussions/136)
- [Project introduction videos](https://github.com/orgs/django-commons/discussions/150)

If any of that sounds interesting to you, feel free to engage on the discussions or here.

**Comment by williln** [comment]
Hi @django-commons/django-fsm-2 -- just doing our semi-annual checkin! If there is anything Django Commons can do to help your project, reach out and let us know. Otherwise, happy coding! 🚢

**Comment by pfouque** [comment]
Hi @williln ,
Everything is going great! 

The only idea that come out of my mind would be to find a way to share a test matrix so that new django version (and its supported python versions) comes automatically...
I'm not sure if and how easy it would be but it would be awesome that project get proactively tested against new versions :)

**Reply by Stormheg** [reply]
That sounds pretty rad to me!

We discussed this during a Django Commons admin meeting and we came to the conclusion there are usually other tasks associated with this as well such as updating trove classifiers in a pyproject.toml file, changelog and readme changes, removing unsupported versions from the matrix and bumping the minimum version constraints

We'd love to hear how other package maintainers tackle this, maybe there is knowledge to be shared.

The first idea that comes to my mind is to let agentic AI do this, it sounds like something it could figure out quite well on its own. Run it on a regular schedule with a prompt telling it what to do and what versions exist – to avoid it hallucinating Django 3.14 as the latest. Have it create a PR that a human can then review and merge; and boom, you've just cut down on one of the maintenance chores.

I probably don't have time to experiment with that, but if someone else is reading this and is inspired: please feel free to experiment with this idea. Might make for an interesting talk topic as well...

**Reply by thibaudcolas** [reply]
I’m 90% likely to work on this as part of Wagtail’s [Package maintainers guide](https://github.com/wagtail/roadmap/issues/108), a revamp of our existing official project template + guidance for maintainers. I wrote a prototype of this to [upgrade Wagtail sites to a new release of Wagtail](https://wagtail.org/blog/an-agent-skill-to-upgrade-your-wagtail-site/), with some degree of support for packages. Also wrote a "upgrade Django" version of this but I didn’t think it was good enough to share.

I’m due to work on this over the next couple of weeks so should have something to share soon™. I don’t think there’ll be much in there that’s specific to Wagtail.

**Comment by tim-schilling** [comment]
Hi @django-commons/django-fsm-2 this is just our semi-annual checkin. Let us know if there is anything Django Commons can do to help, or if there is anything else on your mind! Otherwise, happy coding 🥳

As an additional reminder, though not at all a suggestion/requirement. It may be worth looking through who has regularly contributed to your project to see if you'd like to elevate them to be a committer or a maintainer. I mention it because it's easy to forget to review the repositories community and helping people grow. Similarly, if you or anyone else wants to step back, that's fine too! If we can help coordinate, please let us know.

**Reply by tim-schilling** [reply]
Whoops, forgot some things. There are two other things:

1. Can the @django-commons/django-fsm-2-admins please share your emails with us? You can email us at [django-commons-admin@googlegroups.com](mailto:django-commons-admin@googlegroups.com) to avoid publishing publicly here.
2. We are working on a new policy around project abandonment. As a community organization, we need to have clear guidelines on when the community should pass the project onto someone else or signaling that the project is in a period of low activity. We'll share this new policy with you when we have it ready for comment. This will include a timeline for adoption. If you have questions, please let us know!

**Reply by pfouque** [reply]
Hi @tim-schilling  ,
Everything is still going great!

PS: email shared!

**Reply by tim-schilling** [reply]
Glad to hear it!
